extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.vault;
+
+import java.util.List;
+
+import io.quarkus.vault.transit.ClearData;
+import io.quarkus.vault.transit.DecryptionRequest;
+import io.quarkus.vault.transit.DecryptionResult;
+import io.quarkus.vault.transit.EncryptionRequest;
+import io.quarkus.vault.transit.EncryptionResult;
+import io.quarkus.vault.transit.RewrappingRequest;
+import io.quarkus.vault.transit.Signature;
+import io.quarkus.vault.transit.SigningBatchRequest;
+import io.quarkus.vault.transit.SigningResult;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.Verifiable;
+import io.quarkus.vault.transit.VerificationRequest;
+import io.quarkus.vault.transit.VerificationResult;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    String encrypt(String keyName, ClearData clearData, TransitContext transitContext);
```

Comment:
@vsevel what happens in case of the error in this and the previous `encrypt` ?

Reply:
you get a `VaultException` with the appropriate error message returned from vault. check `VaultTransitBatchResult.getValueOrElseError()`.
For single operations it is the right semantic. For batch operations it is more tricky because as a user you may or may not be interested in the partial result.
For instance when you try to encrypt 1000 values, and get 10 errors, you usually do not want to throw away the 990 encrypted values because 10 failed. Instead you wait a few seconds create a new batch with the just 10 missing encryptions, and call again the encryption service. If you keep having errors after a few tries, and write them somewhere (a file, a kafka topic, ...) for further analysis.
If you have 990 errors, it is a different matter. It means there is really something wrong with your setup. And you should setup the whole process.
This is a very typical way to work with batches.

Reply:
Thanks for it, sure, with the the batch exception all is good IMHO.